library/psa_crypto.c

```diff
@@ -740,22 +731,42 @@ psa_status_t psa_import_key_into_slot( psa_key_slot_t *slot,
 {
     psa_status_t status = PSA_SUCCESS;
 
+    /* zero-length keys are never supported. */
+    if( data_length == 0 )
```

Comment:
I'm willing to take a discussion on this. There's currently a path that would allow zero-length keys to be created (through the key storage API). Should that path exist? I.e. does `psa_save_persistent_key` need a zero-length check as well? Or do you actually want to allow zero-length keys to exist?

Reply:
We do not want zero-length keys. They're specifically forbidden (more trouble than they're worth — a zero-length buffer can have an invalid pointer and that causes undefined behavior in many common circumstances if you don't check the zero-length case which increases code size and complexity). I don't think `psa_save_persistent_key` needs a check since its input has already been validated. A check when loading might be warranted, in case whatever wrote the data had a bug.

Reply:
To confirm, I'm happy with having this check here. At worst, all the code paths leading here now already forbid an empty key and we're wasting a handful of bytes of code size. Having the check improves maintainability in case a code path without an empty-key check gets added later. It also helps static analyzers which might otherwise not be able to realize that an empty key can't happen here.